### ISSUE TITLE
merge_modalities: all-padding sequence causes silent EOS insertion at wrong position

### ISSUE THREAD

**Issue by GerrySant** [GerrySant/multimodalhugs#86]
## Problem

In `multimodalhugs/modules/utils.py`, the EOS token is inserted at the last valid position using:

```python
last_indices = padding_mask.size(1) - torch.argmax(padding_mask.flip(dims=[1]), dim=1) - 1
```

`torch.argmax` on an all-zero (all-padding) vector returns `0` — the first position of the flipped tensor, which maps to the **last** position of the original sequence. This means an all-padding sample silently gets an EOS inserted at its last position rather than raising an error or being caught upstream.

## Why it matters

This function is called on every forward pass when `pad_idx` and `eos_idx` are set (i.e. in all standard training and inference runs). A corrupt sample (empty sequence after filtering failures) will produce a silently malformed batch tensor that feeds into the backbone without any warning. The resulting loss is wrong and debugging is very hard.

## Steps to reproduce

```python
import torch
from multimodalhugs.modules.utils import merge_modalities

# all-padding input: B=1, T=4, D=8
x = torch.zeros(1, 4, 8)
mask = torch.zeros(1, 4, dtype=torch.long)  # all padding
embeddings = torch.nn.Embedding(10, 8)

out_x, out_mask = merge_modalities(x, mask, None, None, embeddings, pad_idx=1, eos_idx=2)
# EOS ends up at position 3 (last), no error raised
```

## Suggested fix

Add a guard before computing `last_indices`:

```python
valid_counts = padding_mask.sum(dim=1)
if (valid_counts == 0).any():
    raise ValueError("merge_modalities received at least one all-padding sequence. "
                     "Check dataset filtering — empty sequences must be removed before batching.")
```

Alternatively, clamp and log a warning rather than raising, depending on the desired robustness level.
